pytorch_lightning/trainer/trainer.py

```diff
@@ -838,6 +836,23 @@ def tune(
 
         return result
 
+    def _restore_checkpoint(self) -> None:
```

Comment:
I believe we need to find a better name for this, as it does not restore the full checkpoint. 
Context: We separated the restoring of model and training state on purpose because it needs to happen at different stages of the whole fit and accelerator setup. 

```suggestion
    def _restore_modules_and_callbacks(self) -> None:
```